platform/logging/LoggingConfig.ts

```diff
@@ -0,0 +1,166 @@
+import { Schema, typeOfSchema } from '../types';
+import { LogLevelId } from './LogLevel';
+import { Appenders, AppenderConfigType } from './appenders/Appenders';
+
+/**
+ * Separator string that used within nested context name (eg. plugins::pid).
+ */
+const CONTEXT_SEPARATOR = '::';
+
+/**
+ * Name of the `root` context that should always be defined.
+ */
+const ROOT_CONTEXT_NAME = 'root';
+
+const createLoggerSchema = (schema: Schema) => {
+  const { literal, object, string, arrayOf, oneOf } = schema;
+
+  return object({
+    level: oneOf(
+      [
+        literal('all'),
+        literal('fatal'),
+        literal('error'),
+        literal('warn'),
+        literal('info'),
+        literal('debug'),
+        literal('trace'),
+        literal('off')
+      ],
+      {
+        defaultValue: 'info'
+      }
+    ),
+    appenders: arrayOf(string(), { defaultValue: [] })
+  });
+};
+
+const createLoggingSchema = (schema: Schema) => {
+  const { object, string, mapOf } = schema;
+
+  return object({
+    appenders: mapOf(string(), Appenders.createConfigSchema(schema), {
+      defaultValue: new Map([
+        [
+          'console',
+          {
+            kind: 'console',
+            layout: {
+              kind: 'pattern',
+              highlight: true
+            }
+          }
+        ]
+      ])
+    }),
+    loggers: mapOf(string(), createLoggerSchema(schema), {
+      defaultValue: new Map([
+        ['root', { appenders: ['console'], level: 'info' as LogLevelId }]
+      ]),
+      validate(value) {
+        const rootLogger = value.get(ROOT_CONTEXT_NAME);
+        if (!rootLogger) {
+          throw new Error(
+            `"${ROOT_CONTEXT_NAME}" logger should be configured!`
+          );
+        }
+
+        if (rootLogger.appenders.length === 0) {
+          throw new Error(
+            `"${ROOT_CONTEXT_NAME}" logger should have at least one "appender" configured.`
+          );
+        }
+      }
+    })
+  });
+};
+
+const loggerConfigType = typeOfSchema(createLoggerSchema);
+/** @internal */
+export type LoggerConfigType = typeof loggerConfigType;
+
+const loggingConfigType = typeOfSchema(createLoggingSchema);
+type LoggingConfigType = typeof loggingConfigType;
+
+/**
+ * Class that describes config used to setup logging subsystem.
+ * @internal
+ */
+export class LoggingConfig {
+  static createSchema = createLoggingSchema;
+
+  /**
+   * Map of the appender unique arbitrary key and its corresponding config.
+   */
+  readonly appenders: Map<string, AppenderConfigType> = new Map();
```

Comment:
I actually decided to get rid of dedicated classes for appender configs, validated config/schema looks good enough for me.